src/test/java/by/a1qa/testreport/ScreenshotListener.java

```diff
@@ -7,25 +7,33 @@
 import org.testng.ITestResult;
 import org.testng.Reporter;
 import org.testng.TestListenerAdapter;
+import ru.yandex.qatools.ashot.AShot;
+import ru.yandex.qatools.ashot.Screenshot;
+import ru.yandex.qatools.ashot.shooting.ShootingStrategies;
 
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
 import java.io.File;
 import java.io.IOException;
-import java.text.SimpleDateFormat;
-import java.util.Calendar;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
 
 public class ScreenshotListener extends TestListenerAdapter {
     @Override
     public void onTestFailure(ITestResult result) {
-        Calendar calendar = Calendar.getInstance();
-        SimpleDateFormat formater = new SimpleDateFormat("dd_MM_yyyy_hh_mm_ss");
-        String dateString = formater.format(calendar.getTime());
+        String dateString = LocalDateTime.now().format(DateTimeFormatter.ISO_DATE_TIME);
         String methodName = result.getName();
         if(!result.isSuccess() && AqualityServices.isApplicationStarted()){
-            File scrFile = ((TakesScreenshot)AqualityServices.getApplication().getDriver()).getScreenshotAs(OutputType.FILE);
+            BufferedImage scrFile = new AShot()
+                    .shootingStrategy(ShootingStrategies.simple())
+                    .takeScreenshot(AqualityServices.getApplication().getDriver()).getImage();
+            File outputFile = new File("image.png");
             try {
+                ImageIO.write(scrFile, "png", outputFile);
                 String reportDirectory = String.format("%s/target/surefire-reports", new File(System.getProperty("user.dir")).getAbsolutePath());
                 File destFile = new File(String.format("%s/failure_screenshots/%s_%s.png", reportDirectory, methodName, dateString));
-                FileUtils.copyFile(scrFile, destFile);
+                FileUtils.copyFile(outputFile, destFile);
```

Comment:
why do you use two files instead of one?

Reply:
Because the second file is used to name the file in the desired format (method name, date)